File: src/features/chrome/chromeUtils.ts
export async function getStoredValues(
  keys: string[]
): Promise<Record<string, unknown>> {
  return new Promise((resolve) => {
    chrome.storage.local.get(keys, (result) => resolve(result));
  });
}

export async function getStoredValue(name: string): Promise<unknown> {
  return new Promise((resolve) => {
    chrome.storage.local.get(name, (result) => resolve(result[name]));
  });
}

export async function setStoredValue(name: string, value: any): Promise<void> {
  return new Promise<void>((resolve) => {
    chrome.storage.local.set({ [name]: value }, () => resolve());
  });
}

File: src/main/kotlin/features/chrome/chromeUtils.kt
package features.chrome

import kotlinx.js.Record
import kotlinx.js.set
import kotlin.js.Promise

fun getStoredValue(name: String): Promise<dynamic> {
    return Promise { resolve, _ ->
        dependencies.chrome.storage.local.get(name) {
            resolve(it[name])
        }
    }
}

fun setStoredValue(name: String, value: Any): Promise<Unit> {
    return Promise { resolve, _ ->
        val items = Record<String, Any>()
        items[name] = value
        dependencies.chrome.storage.local.set(items) {
            resolve(Unit)
        }
    }
}
